framework/include/utils/MooseTypes.h

```diff
@@ -185,6 +185,12 @@ enum LineSearchType
 #endif
 };
 
+enum RegistrationType
+{
+  APPLICATION,
+  OBJECTS
+};
+
```

Comment:
Just wondering why you had a preference for this over just having two different functions...